client_proposal.go

```diff
@@ -6,10 +6,52 @@
 package prnm
 
 import (
+	"crypto/rand"
+	"math/big"
+
+	"perun.network/go-perun/apps/payment"
 	ethwallet "perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/channel"
 	"perun.network/go-perun/client"
+	"perun.network/go-perun/log"
+	"perun.network/go-perun/wallet"
 )
 
+// ProposeChannel proposes a new channel to the given peer (perunID) and with
+// the given parameters. Returns the newly created channel controller if the
+// channel was successfully created and funded.
+//
+// After the channel got successfully created, the user is required to start the
+// update handler with PaymentChannel.HandleUpdates(UpdateHandler) and to start
+// the channel watcher with PaymentChannel.Watch() on the returned channel
+// controller.
+//
+// It is important that the passed context does not cancel before twice the
+// ChallengeDuration has passed (at least for real blockchain backends with wall
+// time), or the channel cannot be settled if a peer times out funding.
+func (c *Client) ProposeChannel(
+	ctx *Context,
+	perunID *Address,
```

Comment:
Hmm yeah we could now use it here. But I'd prefer to do this later if requested. Right now, the doc of the `PeerPerunID` field would also be wrong, because it specifically says that it is the proposing peer. It could also lead to confusion because if A proposes to B, then A calls `ProposeChannel` with B's address as `perunID` and B then receives the request and B's handler gets a `ChannelProposal` where the field is suddenly A's address instead of B's.

Reply:
Then we leave it like this for now.